### ISSUE TITLE
pylint-3 issues

### ISSUE THREAD

**Issue by mjtrangoni** [warpnet/salt-lint#123]
Hi @roaldnefs,

First of all, thanks for this tool. It works very well, and I am excited about how many checks will come in the future.

I tried just for fun running [pylint-3](https://www.pylint.org/) on Fedora 31, and it is actually finding some python issues here and there. Are you interested about this? We can integrate this check to the CI as well.

See,
```bash
$ pylint-3 --disable=missing-docstring,invalid-name $(find . -type f -name "*.py")
************* Module tests
tests/__init__.py:11:0: R0205: Class 'RunFromText' inherits from object, can be safely removed from bases in python3 (useless-object-inheritance)
tests/__init__.py:11:0: R0903: Too few public methods (1/2) (too-few-public-methods)
************* Module tests.unit.TestSkipRule
tests/unit/TestSkipRule.py:54:0: C0305: Trailing newlines (trailing-newlines)
************* Module tests.unit.TestConfig
tests/unit/TestConfig.py:41:44: C0326: No space allowed around keyword argument assignment
        self.config = SaltLintConfig(dict(c = self.fp.name))
                                            ^ (bad-whitespace)
tests/unit/TestConfig.py:75:0: C0305: Trailing newlines (trailing-newlines)
************* Module tests.unit.TestRunner
tests/unit/TestRunner.py:20:0: C0304: Final newline missing (missing-final-newline)
************* Module saltlint.config
saltlint/config.py:22:0: R0902: Too many instance attributes (10/7) (too-many-instance-attributes)
saltlint/config.py:22:0: R0205: Class 'SaltLintConfig' inherits from object, can be safely removed from bases in python3 (useless-object-inheritance)
saltlint/config.py:24:4: W0102: Dangerous default value dict() (builtins.dict) as argument (dangerous-default-value)
saltlint/config.py:40:4: R0912: Too many branches (18/12) (too-many-branches)
saltlint/config.py:22:0: R0903: Too few public methods (1/2) (too-few-public-methods)
saltlint/config.py:5:0: C0411: standard import "import os" should be placed before "import yaml" (wrong-import-order)
saltlint/config.py:6:0: C0411: standard import "import sys" should be placed before "import yaml" (wrong-import-order)
************* Module saltlint.formatters
saltlint/formatters/__init__.py:15:0: R0205: Class 'Formatter' inherits from object, can be safely removed from bases in python3 (useless-object-inheritance)
saltlint/formatters/__init__.py:23:8: R1705: Unnecessary "else" after "return" (no-else-return)
saltlint/formatters/__init__.py:21:4: R0201: Method could be a function (no-self-use)
saltlint/formatters/__init__.py:45:0: R0205: Class 'SeverityFormatter' inherits from object, can be safely removed from bases in python3 (useless-object-inheritance)
saltlint/formatters/__init__.py:53:8: R1705: Unnecessary "else" after "return" (no-else-return)
saltlint/formatters/__init__.py:50:4: R0201: Method could be a function (no-self-use)
saltlint/formatters/__init__.py:78:0: R0205: Class 'JsonFormatter' inherits from object, can be safely removed from bases in python3 (useless-object-inheritance)
saltlint/formatters/__init__.py:80:0: W0613: Unused argument 'args' (unused-argument)
saltlint/formatters/__init__.py:80:0: W0613: Unused argument 'kwargs' (unused-argument)
saltlint/formatters/__init__.py:86:4: R0201: Method could be a function (no-self-use)
************* Module saltlint.cli
saltlint/cli.py:35:0: C0330: Wrong continued indentation (remove 5 spaces).
                           % default_rulesdir)
                      |    ^ (bad-continuation)
saltlint/cli.py:42:0: C0330: Wrong continued indentation (remove 5 spaces).
                           % default_rulesdir)
                      |    ^ (bad-continuation)
saltlint/cli.py:7:0: W0402: Uses of a deprecated module 'optparse' (deprecated-module)
saltlint/cli.py:18:0: R0914: Too many local variables (16/15) (too-many-locals)
saltlint/cli.py:94:7: C1801: Do not use `len(SEQUENCE)` to determine if a sequence is empty (len-as-condition)
saltlint/cli.py:136:4: R1705: Unnecessary "else" after "return" (no-else-return)
saltlint/cli.py:136:7: C1801: Do not use `len(SEQUENCE)` to determine if a sequence is empty (len-as-condition)
saltlint/cli.py:18:0: R0912: Too many branches (14/12) (too-many-branches)
saltlint/cli.py:18:0: R0915: Too many statements (54/50) (too-many-statements)
************* Module saltlint.linter
saltlint/linter.py:60:0: C0330: Wrong continued indentation (add 6 spaces).
                           file['path'], self, message))
                           ^     | (bad-continuation)
saltlint/linter.py:218:0: C0330: Wrong continued indentation (add 15 spaces).
                           skip_list=self.skip_list))
                           ^              | (bad-continuation)
saltlint/linter.py:20:0: R0205: Class 'SaltLintRule' inherits from object, can be safely removed from bases in python3 (useless-object-inheritance)
saltlint/linter.py:26:15: E1101: Instance of 'SaltLintRule' has no 'id' member (no-member)
saltlint/linter.py:26:32: E1101: Instance of 'SaltLintRule' has no 'shortdesc' member (no-member)
saltlint/linter.py:29:15: E1101: Instance of 'SaltLintRule' has no 'id' member (no-member)
saltlint/linter.py:29:32: E1101: Instance of 'SaltLintRule' has no 'shortdesc' member (no-member)
saltlint/linter.py:29:57: E1101: Instance of 'SaltLintRule' has no 'description' member (no-member)
saltlint/linter.py:50:15: E1101: Instance of 'SaltLintRule' has no 'id' member (no-member)
saltlint/linter.py:53:21: E1102: self.match is not callable (not-callable)
saltlint/linter.py:69:18: E1102: self.matchtext is not callable (not-callable)
saltlint/linter.py:77:0: R0205: Class 'RulesCollection' inherits from object, can be safely removed from bases in python3 (useless-object-inheritance)
saltlint/linter.py:95:4: W0102: Dangerous default value set() (builtins.set) as argument (dangerous-default-value)
saltlint/linter.py:147:0: R0205: Class 'Match' inherits from object, can be safely removed from bases in python3 (useless-object-inheritance)
saltlint/linter.py:149:4: R0913: Too many arguments (6/5) (too-many-arguments)
saltlint/linter.py:147:0: R0903: Too few public methods (1/2) (too-few-public-methods)
saltlint/linter.py:162:0: R0902: Too many instance attributes (8/7) (too-many-instance-attributes)
saltlint/linter.py:162:0: R0205: Class 'Runner' inherits from object, can be safely removed from bases in python3 (useless-object-inheritance)
************* Module saltlint.rules.NoTabsRule
saltlint/rules/NoTabsRule.py:17:20: W0613: Unused argument 'file' (unused-argument)
saltlint/rules/NoTabsRule.py:17:4: R0201: Method could be a function (no-self-use)
************* Module saltlint.rules.JinjaPillarGrainsGetFormatRule
saltlint/rules/JinjaPillarGrainsGetFormatRule.py:21:20: W0613: Unused argument 'file' (unused-argument)
saltlint/rules/JinjaPillarGrainsGetFormatRule.py:7:0: C0411: standard import "import re" should be placed before "from saltlint.linter import SaltLintRule" (wrong-import-order)
************* Module saltlint.rules.LineTooLongRule
saltlint/rules/LineTooLongRule.py:20:20: W0613: Unused argument 'file' (unused-argument)
saltlint/rules/LineTooLongRule.py:20:4: R0201: Method could be a function (no-self-use)
************* Module saltlint.rules.YamlHasOctalValueRule
saltlint/rules/YamlHasOctalValueRule.py:20:20: W0613: Unused argument 'file' (unused-argument)
saltlint/rules/YamlHasOctalValueRule.py:7:0: C0411: standard import "import re" should be placed before "from saltlint.linter import SaltLintRule" (wrong-import-order)
************* Module saltlint.rules.NoIrregularSpacesRule
saltlint/rules/NoIrregularSpacesRule.py:41:20: W0613: Unused argument 'file' (unused-argument)
************* Module saltlint.rules.JinjaStatementHasSpacesRule
saltlint/rules/JinjaStatementHasSpacesRule.py:20:20: W0613: Unused argument 'file' (unused-argument)
saltlint/rules/JinjaStatementHasSpacesRule.py:7:0: C0411: standard import "import re" should be placed before "from saltlint.linter import SaltLintRule" (wrong-import-order)
************* Module saltlint.rules.TrailingWhitespaceRule
saltlint/rules/TrailingWhitespaceRule.py:16:20: W0613: Unused argument 'file' (unused-argument)
saltlint/rules/TrailingWhitespaceRule.py:16:4: R0201: Method could be a function (no-self-use)
************* Module saltlint.rules.FileModeLeadingZeroRule
saltlint/rules/FileModeLeadingZeroRule.py:20:20: W0613: Unused argument 'file' (unused-argument)
saltlint/rules/FileModeLeadingZeroRule.py:7:0: C0411: standard import "import re" should be placed before "from saltlint.linter import SaltLintRule" (wrong-import-order)
************* Module saltlint.rules.FileModeQuotationRule
saltlint/rules/FileModeQuotationRule.py:20:20: W0613: Unused argument 'file' (unused-argument)
saltlint/rules/FileModeQuotationRule.py:7:0: C0411: standard import "import re" should be placed before "from saltlint.linter import SaltLintRule" (wrong-import-order)
************* Module saltlint.rules.JinjaCommentHasSpacesRule
saltlint/rules/JinjaCommentHasSpacesRule.py:20:20: W0613: Unused argument 'file' (unused-argument)
saltlint/rules/JinjaCommentHasSpacesRule.py:7:0: C0411: standard import "import re" should be placed before "from saltlint.linter import SaltLintRule" (wrong-import-order)
************* Module saltlint.rules.JinjaVariableHasSpacesRule
saltlint/rules/JinjaVariableHasSpacesRule.py:20:20: W0613: Unused argument 'file' (unused-argument)
saltlint/rules/JinjaVariableHasSpacesRule.py:7:0: C0411: standard import "import re" should be placed before "from saltlint.linter import SaltLintRule" (wrong-import-order)
************* Module saltlint.rules.FileExtensionRule
saltlint/rules/FileExtensionRule.py:20:26: W0613: Unused argument 'line' (unused-argument)
saltlint/rules/FileExtensionRule.py:8:0: C0411: standard import "import os" should be placed before "from saltlint.linter import SaltLintRule" (wrong-import-order)
saltlint/rules/FileExtensionRule.py:1:0: R0801: Similar lines in 2 files
==tests.unit.TestFileModeLeadingZeroRule:13
==tests.unit.TestFileModeQuotationRule:13
testfile:
  file.managed:
    - name: /tmp/testfile
    - user: root
    - group: root
    - mode: '0700'
'''

BAD_MODE_QUOTATION_LINE = '''
testfile:
  file.managed:
    - name: /tmp/badfile
    - user: root
    - group: root (duplicate-code)

------------------------------------------------------------------
Your code has been rated at 8.65/10 (previous run: 8.15/10, +0.50)
```

Let me know if you are interested about this, and I can help you with one/more PR.

**Comment by roaldnefs** [comment]
Thanks @mjtrangoni, I think this would would be a nice improvement of the code quality! Would you be willing to submit a PR to fix the issues listed by `pylint-3` and adding it the Travis CI configuration?

Thanks in advance!

**Comment by roaldnefs** [comment]
Fixed in `master` branch.
